chiselFrontend/src/main/scala/chisel3/core/UserModule.scala

```diff
@@ -170,6 +175,13 @@ abstract class LegacyModule(implicit moduleCompileOptions: CompileOptions)
     names
   }
 
+  private[chisel3] override def namePorts(names: HashMap[HasId, String]): Unit = {
```

Comment:
Does LegacyModule really needs its own specialized code path? I think LegacyModule already checks for extraneous IOs as part of existing code, so suggestName-based IOs should already be rejected. In the case of a suggestName("io") conflict, hopefully the name conflict mechanism should cause it to error out?

Reply:
It probably doesn't, I'll look into it

Reply:
It actually does, the regular UserModule flow involves checking suggestedName and `LegacyModule` needs to not do that. I added a test and then tried merging them but decided to keep them separate